app/grandchallenge/algorithms/views.py

```diff
@@ -495,10 +496,29 @@ def create_upload(image_files):
             ci = interfaces[slug]
             if ci.is_image_kind:
                 if value:
-                    # create civ without image, image will be added when import completes
-                    civ = ComponentInterfaceValue.objects.create(interface=ci)
-                    civs.append(civ)
-                    upload_pks[civ.pk] = create_upload(value)
+                    widget = form.data[f"WidgetChoice-{ci.slug}"]
+                    if widget == WidgetChoices.IMAGE_SEARCH:
+                        (
+                            civ,
+                            created,
+                        ) = ComponentInterfaceValue.objects.get_or_create(
+                            interface=ci, image=value
+                        )
```

Comment:
I'm wondering now if the `get_or_create` calls will cause issues. There might be images for which there are multiple CIVs for a given interface already (the generic interfaces mostly, but maybe also others), right? So I guess this should be out inside a try .. except statement and if multiple civs exist we choose the latest?

Reply:
I've addressed this in the latest commit